src/Between.php

```diff
@@ -162,6 +179,13 @@ public function setInclusive($inclusive)
      */
     public function isValid($value)
     {
+        if ($this->numeric && ! is_numeric($value)) {
+            return false;
+        }
+        if (! $this->numeric && ! is_string($value)) {
+            return false;
+        }
```

Comment:
Also, once you do make this change, add tests for each condition to validate that the expected failure messages are present.